js/views/userPageVw.js

```diff
@@ -322,15 +318,29 @@ module.exports = baseVw.extend({
     });
   },
 
-  bindModalCloseHandler: function(e) {
-    if (this.modalCloseHandler) return;
-
-    this.modalCloseHandler = function() {
-      window.history.back();
+  loadingStatus: function() {
+    var config = {
+      promise: this.loadingDeferred.promise(),
+      cancel: () => {
+        this.userProfileFetch && this.userProfileFetch.abort();
+        this.itemFetch && this.itemFetch.abort();
+      }
     };
+    
+    if (this.currentItemHash) {
+      config.connectText = window.polyglot.t('pageConnectingMessages.listingConnect').replace('${listing}', this.currentItemHash);
+      config.failedText = window.polyglot.t('pageConnectingMessages.listingFail');
+    } else if (this.options.handle) {
+      config.connectText = window.polyglot.t('pageConnectingMessages.handleConnect').replace('${handle}', this.options.handle);
+      config.failedText = window.polyglot.t('pageConnectingMessages.handleFail');
+    } else {
+      config.connectText = window.polyglot.t('pageConnectingMessages.userConnect').replace('${guid}', this.pageID);
+      config.failedText = window.polyglot.t('pageConnectingMessages.userFail');
+      // config.connectTooltip = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed posuere, lectus quis euismod vestibulum, sapien justo laoreet ante, sit amet mollis nibh diam cursus massa. Duis a eros dapibus, ultrices tortor nec, sodales magna.';
```

Comment:
@morebrownies I didn't specify any tooltip text at this time, because I don't know what it should be. I've left the commented out lines as a placeholder.